src/modules/governance/services/governance.abi.service.ts

```diff
@@ -0,0 +1,368 @@
+                totalEnergy: proposal.total_energy.toFixed(),
+                votingDelayInBlocks: proposal.voting_delay_in_blocks.toNumber(),
+                votingPeriodInBlocks: proposal.voting_period_in_blocks.toNumber(),
+                withdrawPercentageDefeated: proposal.withdraw_percentage_defeated.toNumber(),
+            });
+        });
+    }
+
+    @ErrorLoggerAsync({className: GovernanceAbiService.name})
+    @GetOrSetCache({
+        baseKey: 'governance',
+        remoteTtl: CacheTtlInfo.ContractState.remoteTtl,
+        localTtl: CacheTtlInfo.ContractState.localTtl,
+    })
+    async userVotedProposals(scAddress: string, userAddress: string): Promise<GovernanceProposal[]> {
+        return await this.userVotedProposalsRaw(scAddress, userAddress);
+    }
+
+    async userVotedProposalsRaw(scAddress: string, userAddress: string): Promise<GovernanceProposal[]> {
+        const contract = await this.mxProxy.getGovernanceSmartContract(scAddress);
+        const interaction = contract.methods.getUserVotedProposals([userAddress]);
+        const response = await this.getGenericData(interaction);
+
+        return response.firstValue.valueOf().map((proposal: any) => {
+            return new GovernanceProposal({
+
+            });
+        });
+    }
+
+    @ErrorLoggerAsync({className: GovernanceAbiService.name})
+    @GetOrSetCache({
+        baseKey: 'governance',
+        remoteTtl: CacheTtlInfo.ContractState.remoteTtl,
+        localTtl: CacheTtlInfo.ContractState.localTtl,
+    })
+    async proposalVotes(scAddress: string, proposalId: number): Promise<ProposalVotes> {
+        return await this.proposalVotesRaw(scAddress, proposalId);
+    }
+
+    async proposalVotesRaw(scAddress: string, proposalId: number): Promise<ProposalVotes> {
+        const contract = await this.mxProxy.getGovernanceSmartContract(scAddress);
+        const interaction = contract.methods.getProposalVotes([proposalId]);
+        const response = await this.getGenericData(interaction);
+
+        if (!response.firstValue) {
+            return ProposalVotes.default();
+        }
+        const votes = response.firstValue.valueOf();
+        return new ProposalVotes({
+            upVotes: votes.up_votes.toFixed(),
+            downVotes: votes.down_votes.toFixed(),
+            downVetoVotes: votes.down_veto_votes.toFixed(),
+            abstainVotes: votes.abstain_votes.toFixed(),
+            quorum: votes.quorum.toFixed()
+        });
+    }
+
+    @ErrorLoggerAsync({className: GovernanceAbiService.name})
+    @GetOrSetCache({
+        baseKey: 'governance',
+        remoteTtl: CacheTtlInfo.ContractState.remoteTtl,
+        localTtl: CacheTtlInfo.ContractState.localTtl,
+    })
+    async proposalStatus(scAddress: string, proposalId: number): Promise<string> {
+        return await this.proposalStatusRaw(scAddress, proposalId);
+    }
+
+    async proposalStatusRaw(scAddress: string, proposalId: number): Promise<string> {
```

Comment:
you should return a `GovernanceProposalStatus`. Define a EnumType model for graphql

Reply:
done